### ISSUE TITLE
Bitbucket password encryption/decryption with special character (#)

### ISSUE THREAD

**Issue by rdsteele** [capitalone/Hygieia#1091]
I was getting a 401 unauthorized error with the Bitbucket collector. After doing more research, it appears the issue is due to a special character (#) in the password used.

I placed an additional log.debug statement in the DefaultBitbucketServerClient, right after the password decryption: decryptedPassword = Encryption.decryptString(repo.getPassword(), settings.getKey());

password: testpass#
```
2017-01-14 00:50:01,991 DEBUG c.c.d.collector.GitCollectorTask - {password=UKlOFa2bOkMoAJSkMI26bw==, scm=Bitbucket, branch=master, userID=testuser, url=https://git-qa.******.net/scm/devops/testhygieia.git}::master
2017-01-14 00:50:01,992 DEBUG c.c.d.c.DefaultBitbucketServerClient - Rest Url: https://git-qa.******.net/rest/api/1.0/projects/devops/repos/testhygieia/commits?until=master&limit=25
2017-01-14 00:50:01,994 DEBUG c.c.d.c.DefaultBitbucketServerClient - decryptedPassword: testpass
```
As you can see, the '#' is missing.

**Comment by MarkRx** [comment]
I bet the UI isn't base64 encoding the password when sent in. Special characters may thus be interpretted as http parameters.

**Comment by nitin1706** [comment]
I believe this issue is closed with PR #1651
